### PR TITLE
Remove global.yaml

### PR DESCRIPTION
**Pull request by howardjohn**
This fully removes the global.yaml

* Commit 1 and 2: regen operator tests, to ensure we are properly testing this change
* Commit 3:
  * Sync values.yaml to istiod-remote chart
  * Fix a couple missing fields in values.yaml of istiod
  * Remove global.yaml

End result of generate config is identical as seen by the diffs in testing

I highly recommend you review only commit (3), as commits 1 and 2 are auto generated from ./operator/scripts/run_update_golden_snapshots.sh and will be a distraction/

### CODE REVIEW

**Review by jacob-delgado — APPROVED** [review]
lgtm

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -174,6 +174,8 @@ meshConfig:
   # No hurry to do this in 1.6, we're trying to prove the code.
 
 global:
+  # Used to locate istiod.
```
Comment (nrjpoddar) [inline manifests/charts/istio-control/istio-discovery/values.yaml:177]: This is set at the top so not needed again.

File: Makefile.core.mk [hunk 2]
```diff
@@ -347,6 +347,11 @@ gen-check: check-no-modify gen check-clean-repo
 # Copy the injection template file and configmap from istiod chart to istiod-remote chart
 sync-configs-from-istiod:
 	cp manifests/charts/istio-control/istio-discovery/files/injection-template.yaml manifests/charts/istiod-remote/files/
+
+	# Copy over values, but apply some local customizations
+	cp manifests/charts/istio-control/istio-discovery/values.yaml manifests/charts/istiod-remote/
+	yq w manifests/charts/istiod-remote/values.yaml telemetry.enabled false -i
```
Comment (nrjpoddar) [inline Makefile.core.mk:353]: This doesn’t seem maintainable. Is there a better way to provide/merge overrides.

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 3]
```diff
@@ -482,3 +484,5 @@ global:
 
   # Use the Mesh Control Protocol (MCP) for configuring Istiod. Requires an MCP source.
   useMCP: false
+
+istiodRemote: {}
```
Comment (nrjpoddar) [inline manifests/charts/istio-control/istio-discovery/values.yaml]: Why is this needed?

File: Makefile.core.mk (hunk 2 above)
Comment (howardjohn) [reply Makefile.core.mk:353]: There is `yq merge` but I could n't get it working

File: manifests/charts/istio-control/istio-discovery/values.yaml (hunk 3 above)
Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/values.yaml]: Otherwise the helm chart fails since this value is referencing in the chart

Comment (nrjpoddar) [reply manifests/charts/istio-control/istio-discovery/values.yaml]: I was able to install/upgrade without this value so I’m guessing it’s referenced under a conditional?

File: Makefile.core.mk (hunk 2 above)
Comment (nrjpoddar) [reply Makefile.core.mk:353]: I just meant how will anyone know which values to merge/override?

Comment (howardjohn) [reply Makefile.core.mk:353]: I don't know, any suggestions? This is no different than current code btw, just more explicit. we probably need to clean up the charts given the new multicluster structure

Comment (nrjpoddar) [reply Makefile.core.mk:353]: Not really. It's better than having two full copies I guess.

**Review by nrjpoddar — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/values.yaml (hunk 1 above)
Comment (nrjpoddar) [reply manifests/charts/istio-control/istio-discovery/values.yaml:180]: @howardjohn can you remove this or the same option set on line 202? Thanks!

**Comment by nrjpoddar** [comment]
@howardjohn lets get this merged as without it in 1.8 Helm support will be fragile with globals.yaml around, thanks!

**Review by ericvn — APPROVED** [review]

**Review by ostromart — APPROVED** [review]

**Comment by nrjpoddar** [comment]
@howardjohn can you cherry-pick for 1.8 or is it too late now?

**Comment by howardjohn** [comment]
> @howardjohn can you cherry-pick for 1.8 or is it too late now?

For 1.8.0 its too late I think. For patch release.. I think this would be a breaking change to some extent? Perhaps we can just make it so global.yaml is there but optional? Which would involve a partial backport of just the few lines that changed in the charts?

**Comment by nrjpoddar** [comment]
> > @howardjohn can you cherry-pick for 1.8 or is it too late now?
> 
> For 1.8.0 its too late I think. For patch release.. I think this would be a breaking change to some extent? Perhaps we can just make it so global.yaml is there but optional? Which would involve a partial backport of just the few lines that changed in the charts?

That works for me. We don’t document using `globals.yaml’ anywhere afaik for users so I’m not sure if it’s even a breaking change for patch releases but if you want to conservative I’m good with it.
